### ISSUE TITLE
[Feature] Add ability to specify video params (like fps) for video recordings

### ISSUE THREAD

**Issue by rachitm022** [microsoft/playwright#17217]
Would be good to specify params for video recordings like fps 

```js
const context = await browser.newContext({
  recordVideo: {
    dir: 'videos/',
    size: { width: 640, height: 480 },
    fps: 25
  }
});
```

**Comment by HeDo88TH** [comment]
This is a missing piece, we need this.

**Comment by riethmue** [comment]
Hi, I need this feature. When do you plan to roll it out? :blush:

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.

**Comment by terion-labs** [comment]
years pass...

**Comment by angel1254mc** [comment]
This would be awesome. Our team has been experimenting with using Playwright to automate demo recordings.  The idea is to generate scripted browser walkthroughs that could be regenerated with every release. Configurable FPS would let us produce smoother, more professional output for these kinds of non-test use cases.

Totally acknowledge that Playwright is at the mercy of the browser's devtools protocol when it comes to the count/timing of frames it receives, so we can't really guarantee that a user's desired FPS will be respected.

One approach could be adding an optional `fps` param to `recordVideo` that lets users set a _target_ fps for the output video (defaults to today's 25 fps). If the browser sends enough frames, awesome. Otherwise the user just gets repeated frames to fill the gap, which matches how it works today at the hardcoded 25 fps. Thoughts?

**Comment by MuTsunTsai** [comment]
Hi @rachitm022, @angel1254mc, @terion-name — for the FPS / demo recording use case described here, [`playwright-recorder-plus`](https://github.com/MuTsunTsai/playwright-recorder-plus) might be useful:

- `fps` option (default 25), pluggable via `attachRecorder(page, { fps: 60 })`.
- **@angel1254mc**: your "scripted browser walkthroughs that could be regenerated with every release" is exactly the niche I built this for. CDP frame cadence varies as you noted, so the recorder runs same-slot dedup + previous-frame padding + tail padding to keep the encoded video aligned with wall-clock regardless of CDP gaps.
- Supports pause/resume, crop, multi-page contexts, and inline audio scheduling for narration / click sounds.

Wraps Playwright 1.59+'s public `page.screencast` API + ffmpeg-static, no patching needed. Not a Playwright maintainer; sharing in case it unblocks anyone.

**Comment by azmeuk** [comment]
Would like to widen the scope of this issue to cover the sibling encoding parameters that share the same pipeline:

- codec - VP8 (current default) / VP9
- quality or bitrate / CRF

My use case is generating screencasts for library documentation (with [sphinxcontrib-screencast](https://github.com/tushuhei/sphinxcontrib-screenshot), which drives Playwright in headless). The default VP8 encoding produces artefacts on small monospaced text in code blocks so being able to tweak the parameters would help. Also, tweaking would allow to produce smaller artefacts.

Suggested shape, extending the original example:

```js
recordVideo: {
  dir: 'videos/',
  size: { width: 640, height: 480 },
  fps: 25,
  codec: 'vp9',
  quality: { mode: 'crf', value: 23 },
}
```
